### PR TITLE
deletion of bastion blocked by ipv6

### LINKED ISSUE #376: Bastions are not cleaned up when using ipv6
**How to categorize this issue?**

/area robustness
/kind bug
/platform gcp

**What happened**:
I created a bastion containing an IPv6 address, knowing that it is not supported for gcp. After I saw the expected error in the status, I marked the bastion for deletion.

```yaml
apiVersion: operations.gardener.cloud/v1alpha1
kind: Bastion
metadata:
  annotations:
    gardener.cloud/created-by: me@example.com
  creationTimestamp: "2022-01-14T13:44:58Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2022-01-14T13:45:03Z"
  finalizers:
  - gardener
  generation: 2
  name: cli-31pyggag
  namespace: garden-myproject
  ownerReferences:
  - apiVersion: core.gardener.cloud/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Shoot
    name: my-cluster
    uid: 6b565506-539a-410e-ac3d-b1b9f313fab4
  resourceVersion: "1658557427"
  uid: 9f2c57a5-5789-4de9-b5fc-ee3679349073
spec:
  ingress:
  - ipBlock:
      cidr: 1c01:8071:7a1:aa00::/64
  - ipBlock:
      cidr: 1.2.3.4/32
  providerType: gcp
  seedName: gcp
  shootRef:
    name: my-cluster
  sshPublicKey: ssh-rsa AAAAB3...
status:
  conditions:
  - lastTransitionTime: "2022-01-14T13:45:14Z"
    lastUpdateTime: "2022-01-14T13:45:14Z"
    message: The bastion is being deleted.
    reason: DeletionInProgress
    status: "False"
    type: BastionReady
  expirationTimestamp: "2022-01-14T14:44:58Z"
  lastHeartbeatTimestamp: "2022-01-14T13:44:58Z"
```

**What you expected to happen**:
That the bastion resource is finally deleted. This is not the case, instead, `func ingressPermissions(bastion *extensionsv1alpha1.Bastion) ([]string, error)`, respectively `func DetermineOptions(bastion *extensionsv1alpha1.Bastion, cluster *controller.Cluster, projectID string) (*Options, error) {`
just returns with an error.

https://github.com/gardener/gardener-extension-provider-gcp/blob/master/pkg/controller/bastion/options.go#L149

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version (if relevant):
- Extension version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by tedteng**
**How to categorize this PR?**
/platform gcp

**What this PR does / why we need it**:
 fix deletion blocked if ipv6 apply

**Which issue(s) this PR fixes**:
Fixes #376

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

**Review by petersutter — DISMISSED** [review]
This PR solves the issue pragmatically, however I would rather get rid of the `CIDRs` field on the options and instead pass the cidr to the functions that require the cidrs value (`IngressAllowSSH`, `patchCIDRs`). In addition I would move the `ingressPermissions` function to the `actuator_reconile.go` file, as it is now only used there.
I leave that up to you / the maintainers to decide

/lgtm

**Comment by rfranzke** [comment]
@tedteng What do you plan to do about @petersutter's comment?

**Comment by tedteng** [comment]
sorry, miss the message and holiday last week. will have a look

**Comment by tedteng** [comment]
commit per your comments @petersutter

File: pkg/controller/bastion/actuator_reconcile.go [hunk 1]
```diff
@@ -132,26 +132,24 @@ func ensureFirewallRules(ctx context.Context, gcpclient gcpclient.Interface, bas
...
 
-	firewall, err := getFirewallRule(ctx, gcpclient, opt, IngressAllowSSH(opt).Name)
+	firewall, err := getFirewallRule(ctx, gcpclient, opt, IngressAllowSSH(opt, cidrs).Name)
 	if err != nil || firewall == nil {
 		return fmt.Errorf("could not get firewall rule: %w", err)
 	}
 
 	currentCIDRs := firewall.SourceRanges
-	wantedCIDRs := opt.CIDRs
+	wantedCIDRs := cidrs
```
Comment (petersutter) [inline pkg/controller/bastion/actuator_reconcile.go:149]: nit: you could have also already name it `wantedCIDRs` at the top
```go
func ensureFirewallRules(ctx context.Context, gcpclient gcpclient.Interface, bastion *extensionsv1alpha1.Bastion, opt *Options) error {
	wantedCIDRs, err := ingressPermissions(bastion)
```

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]

**Pull request merged**